### PR TITLE
Remove outbound calls to KeyVault for Prod and Staging

### PR DESCRIPTION
**Pull request by rekhaswaminathan**
Created separate branch for removing outbound keyvault calls. I will merge this branch to main once the devops pipeline is integrated with the [script ](https://github.com/Azure/Azure-AppServices-Diagnostics/pull/598)and tested. 

- For production and staging, skip outbound calls 

- @hforeste , Can you review this for N-clouds. From your email I learnt that N-clouds is using keyvault app setting reference so it shouldn't be affected.

### CODE REVIEW

File: src/Diagnostics.DataProviders/ConfigurationFactory.cs [hunk 1]
```diff
@@ -32,15 +32,19 @@ public AppSettingsDataProviderConfigurationFactory(IHostingEnvironment env)
...
-            var keyVaultClient = new KeyVaultClient(
-                new KeyVaultClient.AuthenticationCallback(
-                    tokenProvider.KeyVaultTokenCallback
-                )
-            );
-
-            string keyVaultConfig = Helpers.GetKeyvaultforEnvironment(env.EnvironmentName);
-            builder.AddAzureKeyVault(builtConfig[keyVaultConfig], keyVaultClient, new DefaultKeyVaultSecretManager());
+            // For production and staging, skip outbound call to keyvault for AppSettings
+            if (env.IsDevelopment())
```
Comment (hforeste) [inline src/Diagnostics.DataProviders/ConfigurationFactory.cs:36]: Should we also use the same boolean condition that you have in startup code? `builtConfig.GetValue<bool>("Secrets:KeyVaultEnabled", true) || hostingEnvironment.IsDevelopment()`

Comment (rekhaswaminathan) [reply src/Diagnostics.DataProviders/ConfigurationFactory.cs:36]: `Secrets:KeyVaultEnabled` - this config is present only in CompilerHost. So thats why I took this approach for runtimehost - If development include keyvault else skip the call.

Comment (hforeste) [reply src/Diagnostics.DataProviders/ConfigurationFactory.cs:36]: Okay. Just quick clarification for N-Clouds, I am using keyvault app settings reference specifically for observer in N-Clouds. Haven't transitioned to using that for runtimehost. You think we can include the `KeyVaultEnabled` flag as part of your condition? Then I'll just keep it enabled like the old way in N-clouds until I move to using keyvault appsetting references or the new approach we're taking in public.

Comment (rekhaswaminathan) [reply src/Diagnostics.DataProviders/ConfigurationFactory.cs:36]: Okay sounds good! Thanks for clarification @hforeste

**Review by hforeste — APPROVED** [review]

**Review by ShekharGupta1988 — APPROVED** [review]

**Pull request merged**